src/object_store/factory.rs

```diff
@@ -134,16 +134,28 @@ impl ObjectStoreFactory {
             }
         };
 
+        // Any path provided in the url has not been included in the object store root, so it
+        // needs to become a part of the prefix, alongside with the actual table name (unless
+        // it's a file/memory store)
+        let prefix = if !url.path().is_empty()
+            && url.scheme() != "file"
+            && url.scheme() != "memory"
+        {
+            format!("{}/{table_path}", url.path())
+        } else {
+            table_path.clone()
+        };
+
+        // This is the least surprising way to extend the path, and make the url point to the table
+        // root: https://github.com/servo/rust-url/issues/333
+        url.path_segments_mut().unwrap().push(&table_path);
```

Comment:
`Url::join` is unreliable/surprising, so it actually will swallow up the prefix, whereas this works.

Reply:
can this unwrap() be easily replaced by an error?

Reply:
Yup, replaced.